src/ClientGenerator/ClientGenerator.cs

```diff
@@ -105,15 +106,23 @@ private bool CreateGrainClient(CodeGenOptions options)
                 Directory.CreateDirectory(outputFileDirectory);
             }
 
+            var services = new ServiceCollection();
+            services.AddSingleton<AssemblyProcessor>();
+            services.AddSingleton(sp => sp);
+            services.AddSingleton<IMessagingConfiguration>(new GlobalConfiguration());
+            services.AddSingleton<RoslynCodeGenerator>();
+            services.AddSingleton<SerializationManager>();
+            var serviceProvider = services.BuildServiceProvider();
+            var codeGenerator = serviceProvider.GetRequiredService<RoslynCodeGenerator>();
```

Comment:
Seems peculiar to construct a container to construct a class rather than just constructing the class directly.